### ISSUE TITLE
integration flake: TestPersistentVolumeMultiPV

### ISSUE THREAD

**Issue by wojtek-t** [kubernetes/kubernetes#27179]
```
=== RUN   TestPersistentVolumeMultiPVs
--- FAIL: TestPersistentVolumeMultiPVs (10.96s)
	persistent_volumes_test.go:254: volumes created
	persistent_volumes_test.go:260: claim created
	persistent_volumes_test.go:264: volume bound
	persistent_volumes_test.go:266: claim bound
	persistent_volumes_test.go:284: Bind mismatch! Expected pvc-2 capacity 50000000000 but got fake-pvc-57 capacity 5000000000
```

https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/27176/kubernetes-pull-test-unit-integration/30016/artifacts/

@jsafrane

**Comment by jsafrane** [comment]
>   persistent_volumes_test.go:284: Bind mismatch! Expected pvc-2 capacity 50000000000 but got fake-pvc-57 capacity 5000000000

That does not make any sense. `fake-pvc-57` should not exist at this point, it's from different integration test and `TestPersistentVolumeMultiPVs` starts with `deleteAllEtcdKeys()`.

@ixdy, Is it possible that that another test is running at this time? Each test basically starts with `deleteAllEtcdKeys()`, so parallelism is IMO not possible as the tests will delete keys of each other.

**Comment by jsafrane** [comment]
So, it looks like the tests are not running in parallel, however some volumes leak from one test to another one, even past `deleteAllEtcdKeys()`. I have no explanation for this, but I can reproduce it locally with logs I need.

I have gut feeling that there is something wrong in the way how controller stops - each test starts a new one and stops it in the end. `PV controller.Stop()` just closes some channels, there is no way how to wait until `Reflector.RunUntil(stopCh)` and `Controller.processLoop(stopCh)` finishes.

I'll look at it on Monday, any ideas would be welcome.

**Comment by ixdy** [comment]
The tests might be running in parallel - I think that's what `go test` does by default.

**Comment by ixdy** [comment]
though that parallelism may only be across separate test suites, so `test/integration` is probably not run in parallel.

**Comment by dchen1107** [comment]
I bumped the priority to P0 since it is the main cause for kubernetes-test-go now, which blocks merge queue.

**Comment by lavalamp** [comment]
This issue is dropping our merge throughput by 60%...

**Comment by lavalamp** [comment]
I've made a quick patch which will at least test @jsafrane's theory. I'll post it shortly.

**Comment by lavalamp** [comment]
Hm, now I see a failure in integration.TestUnschedulableNodes-- will keep my eye on it

**Comment by wojtek-t** [comment]
There is an open issue for UnschedulableNodes - @mml FYI

**Comment by jsafrane** [comment]
@lavalamp, thanks for the PR. Did the test fail again after #27255 was merged?

**Comment by jsafrane** [comment]
I still see some failures of `TestPersistentVolume*` on my machine, fixed by #27271
